dts/arm/st/h5/stm32h562.dtsi

```
@@ -328,6 +328,22 @@
 				status = "disabled";
 			};
 		};
+	
+		timers8: timers@40013400 {
+			compatible = "st,stm32-timers";
+			reg = <0x40013400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00002000>;
+			resets = <&rctl STM32_RESET(APB2, 13U)>;
```

```suggestion
			resets = <&rctl STM32_RESET(APB2, 13)>;
```

---

But it looks like the standard in this device tree is unsigned literals for the values?

---

"Unsigned literals" is a C concept - the Device Tree specification defines all integer properties as unsigned.

`dtlib`, the library used by Zephyr to parse DTS files, will accept such numbers but silently discards the `U` suffix:
https://github.com/zephyrproject-rtos/zephyr/blob/892e5e11face678a7cb5888f4a273c2e52421b46/scripts/dts/python-devicetree/src/devicetree/dtlib.py#L2115
https://github.com/zephyrproject-rtos/zephyr/blob/892e5e11face678a7cb5888f4a273c2e52421b46/scripts/dts/python-devicetree/src/devicetree/dtlib.py#L1570-L1577

The presence of `U`-suffixed numbers across STM32 DTS is an unfortunate artifact of the PR that introduced the `STM32_CLOCK` macro (c.f. [my comment](https://github.com/zephyrproject-rtos/zephyr/pull/79683#pullrequestreview-2366609352)), but it should be avoided for new series or nodes.

---

Got it. Thanks for the explanation!

---

> The presence of U-suffixed numbers across STM32 DTS is an unfortunate artifact of the PR that introduced the STM32_CLOCK macro (c.f. https://github.com/zephyrproject-rtos/zephyr/pull/79683#pullrequestreview-2366609352), but it should be avoided for new series or nodes.

We should clean this up (in another PR for sure).